### PR TITLE
Renaming a file should re-add the extension if missing 11903

### LINKED ISSUE #11903: Renaming a file should re-add the extension if missing
**Is your suggestion for improvement related to a problem? Please describe.**
When I rename a file, the extension is not kept

**Describe the solution you'd like**

1. I add a file to an entry.  I want to rename the file For that, I press Ctrl+R
2. The rename dialog appears - the whole filename is selected:\
    image
3. I can type a new file name (e.g., `demo`
4. I press <kbd>Enter</kbd>

- Expected result: The file is named `demo.pdf`
- Actual result: The file is named `demo` (without extension)

JabRef should check whether the user removed the extension. If JabRef cannot find a new extension, it should put the old one (and not just keep the file as is)

Other implementation possibility: The dialog should select the filename only - without extension.

### PR DESCRIPTION
**Pull request by ivobubenko**
In the linkedFileHandler, the function renameToName now stores extension from old filename and uses it if the new name does not provide one. 

Closes #11903 

### Mandatory checks

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: CHANGELOG.md
Comment (subhramit) [inline CHANGELOG.md:60]:
```suggestion
- When renaming a file, the old extension is now used if there is none provided in the new name. [#11903](https://github.com/JabRef/jabref/issues/11903)
```

File: src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java
Comment (subhramit) [inline src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java:90]: Is this tested? Does `getFileExtension(...)` return null when there is no extension provided? Or an empty string?
Because `isPresent()` returns false only for null values.

**Review by subhramit — DISMISSED** [review]

File: src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java
Comment (subhramit) [reply src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java:90]: Also try out this case: what happens if renamed filename = `f1.` (trailing dot but no extension).

File: src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java
Comment (koppor) [inline src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java:91]: Please really use the power of optionals.

Only go into the "modification" path if `FileUtil.getFileExtension` is present.

Needs some more thinking with Optionals...

The curren tcode path `test.` as possible new file name won't work.

Please also add test cases. You can use `@TempDir` of JUnit to create fies.  It is a bit more effort, but we need that to ensure the code works.

**Review by koppor — CHANGES_REQUESTED** [review]

File: src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java
Comment (ivobubenko) [reply src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java:91]: I am unable to find LinkedFileHandlerTest class and externalfiles folder at logic tests, does this mean I should create them or there is another test class where the test for renaming should be implemented?

Comment (Siedlerchr) [reply src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java:91]: I would create an own test class for it. Press ctrl (cmd on mac) + shift + T in intellj to create a new test class
You can take a look at LinkedFileViewModelTest for similar tests

File: src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java
Comment (ivobubenko) [reply src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java:90]: I am currently writing tests for this functionality. Right now the result of renaming file: **f1.pdf** to f1. is **f1..pdf**, if this is not the expected result, please let me know what the result should be so I can implement tests correctly.

Comment (subhramit) [reply src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java:90]: Yes, this is alright!

File: src/test/java/org/jabref/logic/externalfiles/LinkedFileHandlerTest.java
Comment (Siedlerchr) [inline src/test/java/org/jabref/logic/externalfiles/LinkedFileHandlerTest.java:47]: Why do you need a cookie handler in the LinkedFilesTest?  It's not needed here, you don't do any downloads

File: src/test/java/org/jabref/logic/externalfiles/LinkedFileHandlerTest.java
Comment (koppor) [inline src/test/java/org/jabref/logic/externalfiles/LinkedFileHandlerTest.java:48]: The tests are much duplicated code lines. Please use `@ParameterizedTest`.

File: src/test/java/org/jabref/logic/externalfiles/LinkedFileHandlerTest.java
Comment (koppor) [inline src/test/java/org/jabref/logic/externalfiles/LinkedFileHandlerTest.java:68]: No explanation test for `assertEquals`. The name of the test method and the excpected string should be enough to know what's going on.

**Review by koppor — CHANGES_REQUESTED** [review]

File: src/test/java/org/jabref/logic/externalfiles/LinkedFileHandlerTest.java
Comment (koppor) [reply src/test/java/org/jabref/logic/externalfiles/LinkedFileHandlerTest.java]: Not sure whether some AI was used here. - Hints on AI usage can be found at https://roe.dev/blog/using-ai-in-open-source.

File: src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java
Comment (koppor) [inline src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java]: It should still be possible to rename files: E.g., `startUp` to `start-up`. (Even though this is an edge case)

**Review by koppor — APPROVED** [review]
I fixed the logic for files without extension - and made the logic more readable (IMHO)

The changes need a second review and thus this PR awaits a second review.

**Review by calixtus — APPROVED** [review]

**Pull request merged**
